pkg/handlers/file_uploads.go

```diff
@@ -24,32 +42,132 @@ func NewFileUploadHandler(base HandlerBase, createURL CreateFileUploadURL) FileU
 type FileUploadHandler struct {
 	HandlerBase
 	CreateFileUploadURL CreateFileUploadURL
+	CreateUploadedFile  CreateUploadedFile
+	FetchUploadedFile   FetchUploadedFile
 }
 
 // Handle handles a request for file uploading
 func (h FileUploadHandler) Handle() http.HandlerFunc {
```

Comment:
A question about authz/authn... Is this endpoint open to anyone, or is the auth done at a different layer (maybe middleware)?

Reply:
For posterity, we talked about this in #583.